x-pack/plugins/viz_editor/common/query_types.ts

```diff
@@ -8,13 +8,24 @@
 // Select clause
 // ----------------------------------------
 
+export type OperationName = 'column' | 'count' | 'date_histogram' | 'sum' | 'avg';
+
 export interface Aliasable {
+  operation: OperationName;
   alias?: string;
 }
 
-export interface Column extends Aliasable {
-  operation: 'col';
-  argument: string;
+export interface GenericOperation extends Aliasable {
```

Comment:
Naming is hard...

The name`GenericOperation` makes me expect it to use generics (e.g. `GenericOperation<Foo>`).

Basically, this is an intersection type (the intersection of aliasable and fieldable?) 

```ts
interface Aliasable {
  operation: SelectOperator;
  alias?: string;
}

interface Fieldable {
  argument: Field;
}

type AliasableField = Fieldable & Aliasable;
```

Incidentally, this is one of the things I like least about statically typed programming. You have an explosion of names, and naming is one of the "hard problems" in computer science. Not sure how to minimize this...

Reply:
You're right, using an intersection makes more sense here. It might also satisfy more of the constraints around this.